lib/cn/node_split.c

```diff
@@ -133,27 +166,227 @@ cn_tree_node_get_split_key(
         goto out;
 
     while (bin_heap2_pop(bh, (void **)&kblk)) {
-        wlen += kblk->kb_metrics.tot_key_bytes + kblk->kb_metrics.tot_val_bytes;
-        if (wlen >= total_wlen / 2)
+        kvlen += kblk->kb_metrics.tot_key_bytes + kblk->kb_metrics.tot_val_bytes;
+        if (kvlen >= total_kvlen / 2)
             break;
     }
 
-    if (key_buf) {
-        const size_t copy_len = kblk->kb_klen_min < key_buf_sz ? kblk->kb_klen_min : key_buf_sz;
+    *seen_kvlen = kvlen;
+
+    /* Gather the offsets we ended at for each kvset. These will seed the
+     * starting positions of the element sources in the next bin heap.
+     */
+    for (uint64_t i = 0; i < num_kvsets; i++)
+        offsets[i] = iters[i]->offset;
+
+out:
+    if (buf) {
+        for (uint64_t i = 0; i < num_kvsets; i++)
+            reverse_kblk_iterator_destroy(iters[i]);
+    }
+
+    free(buf);
+    bin_heap2_destroy(bh);
+
+    return 0;
```

Comment:
return err;

Reply:
Good catch. Fixed in the other function as well.